crates/ruff_linter/src/rules/pylint/rules/unnecessary_dunder_call.rs

```diff
@@ -156,8 +157,45 @@ pub(crate) fn unnecessary_dunder_call(checker: &mut Checker, call: &ast::ExprCal
         call.range(),
     );
 
-    if let Some(fixed) = fixed {
-        diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(fixed, call.range())));
+    if let Some(mut fixed) = fixed {
+        let is_in_unary = checker
+            .semantic()
+            .current_expression_parent()
+            .is_some_and(|parent| matches!(parent, Expr::UnaryOp(ast::ExprUnaryOp { .. })));
+
+        if is_in_unary {
+            // find the first ")" before our dunder method
+            let rparen =
+                SimpleTokenizer::starts_at(value.as_ref().end(), checker.locator().contents())
```

Comment:
Can we start after the `attribute` to reduce the text that need to be searched?

Reply:
Can't start after the value in this case.

We're looking for this,

```
(-a).__sub__(1)
###^
```

and the difference between

`(-a).__sub__(1)` and `-a.__sub__(1)`

It could possibly be optimized another way though, perhaps by checking the tokens between `value` and the first `Dot` 🤔